### PR TITLE
Drop `core-js` and move to vite

### PR DESCRIPTION
**Pull request by susnux**
* Move from babel to Vite for bundling and use our config to be consistent
* Drop `core-js` as a dependency, it is automatically included and loaded by server core.

### CODE REVIEW

**Review by ShGKme — APPROVED** [review]
I haven't checked what we do in other libs, but this migration brings one change - dist is now minimized.

I'm not sure it is needed because:
- The final build of app/library is handled by their bundling anyway, minimizing library dist doesn't have any advantages
- It makes debug of apps/libs more complicated, because even in dev build of app/lib, this dependency is minimized

Other than that, everything is fine

File: package.json [hunk 1]
```diff
@@ -2,41 +2,45 @@
...
+      "require": "./dist/index.cjs"
+    }
+  },
+  "scripts": {
+    "build": "vite --mode production build",
+    "build:doc": "typedoc --out dist/doc lib && touch dist/doc/.nojekyll",
+    "dev": "vite --mode development build",
+    "dev:watch": "vite --mode development build --watch",
+    "lint": "eslint . lib",
+    "lint:fix": "eslint --fix lib",
```
Comment (ShGKme) [inline package.json:30]:
```suggestion
    "lint": "eslint lib",
    "lint:fix": "eslint --fix lib",
```

Comment (susnux) [reply package.json]: :white_check_mark:

**Comment by susnux** [comment]
> I haven't checked what we do in other libs, but this migration brings one change - dist is now minimized.

We do it everywhere as well. Maybe adjust at `@nextcloud/vite-config`.

**Pull request merged**